portal/src/graphql/portal/AuthenticationCountryCallingCodeList.tsx

```diff
@@ -0,0 +1,490 @@
+    // setting a large distance means location constraint will be
+    // satisfied no matter where the match is located
+    distance: 10000,
+  }
+);
+
+function makeCountryCodeListColumns(
+  renderToString: (messageId: string) => string
+): IColumn[] {
+  return [
+    {
+      key: "selected",
+      fieldName: "selected",
+      name: renderToString("AuthenticationWidget.phone.list-header.active"),
+      minWidth: 90,
+      maxWidth: 90,
+      className: styles.callingCodeListColumn,
+    },
+    {
+      key: "countryName",
+      fieldName: "countryName",
+      name: renderToString(
+        "AuthenticationWidget.phone.list-header.country-or-area"
+      ),
+      minWidth: 240,
+      maxWidth: 240,
+      isMultiline: true,
+      className: cn(styles.countryNameCell, styles.callingCodeListColumn),
+    },
+    {
+      key: "callingCode",
+      fieldName: "callingCode",
+      name: renderToString("AuthenticationWidget.phone.list-header.code"),
+      minWidth: 65,
+      maxWidth: 65,
+      className: styles.callingCodeListColumn,
+    },
+    {
+      key: "order",
+      name: renderToString("AuthenticationWidget.phone.list-header.order"),
+      fieldName: "order",
+      minWidth: 175,
+      maxWidth: 175,
+      className: styles.callingCodeListColumn,
+    },
+  ];
+}
+
+// asusume country calling code data has no duplicated code
+function constructCallingCodeListData(
+  allCountryCallingCodes: string[],
+  getTelecomCountryName: (code: string) => string
+): CountryCallingCodeListData {
+  const callingCodeListData: CountryCallingCodeListData = {};
+  for (const callingCode of allCountryCallingCodes) {
+    const countryCodes =
+      countryCallingCodeMap[callingCode as keyof typeof countryCallingCodeMap];
+    const countryName = countryCodes
+      .map((code) => getTelecomCountryName(code))
+      .join(", ");
+
+    callingCodeListData[callingCode] = {
+      key: callingCode,
+      countryName,
+      callingCode,
+    };
+  }
+  return callingCodeListData;
+}
+
+function constructCallingCodeListItem(
+  allCountryCallingCodes: string[],
+  selectedCountryCallingCodes: string[],
+  callingCodeListData: CountryCallingCodeListData,
+  searchString: string
+): CountryCallingCodeListItem[] {
+  const selectedCountryCallingCodeSet = new Set(selectedCountryCallingCodes);
+  const inputUnselectedCountryCallingCodes = allCountryCallingCodes.filter(
+    (callingCode) => !selectedCountryCallingCodeSet.has(callingCode)
+  );
+
+  let rawUnselectedCodeList: string[] = inputUnselectedCountryCallingCodes;
+  if (searchString.trim() !== "") {
+    const matchedCallingCodeItems = fuseSearcher.search(searchString);
+    const matchedUnselectedCodes = matchedCallingCodeItems
+      .filter((item) => !selectedCountryCallingCodeSet.has(item.callingCode))
+      .map((item) => item.callingCode);
+    rawUnselectedCodeList = matchedUnselectedCodes;
+  }
+
+  const unselectedCodeList = rawUnselectedCodeList.sort((code1, code2) => {
```

Comment:
FYI, a more idiomatic way to write comparer function: `(code1, code2) => Number(code1) - Number(code2)`.

Reply:
Thanks, this looks much cleaner